src/MyPlot/EventListener.php

```diff
@@ -270,7 +278,12 @@ public function onPlayerMove(PlayerMoveEvent $event) {
 				$popup = TextFormat::WHITE . $paddingPopup . $popup . "\n" .
 					TextFormat::WHITE . $paddingOwnerPopup . $ownerPopup;
 			}
-			$event->getPlayer()->sendTip($popup);
+			if ($this->plugin->getConfig()->get("ShowPlotPopup", true))
+				$event->getPlayer()->sendTip($popup);
+		}
+
+		if ($plotFrom !== null and ($plot === null or $plot !== $plotFrom)) {
+			$this->plugin->getServer()->getPluginManager()->callEvent($ev = new MyPlotPlayerLeavePlotEvent($this->plugin, $event->getPlayer()->getName(), $plotFrom, $event->getPlayer()));
```

Comment:
Could you make this cancellable too for further expandability?

Reply:
No. It would create loops in cancelling movement if entering a plot is cancelled.

Reply:
If the event for entering a plot is called, the one opposite shouldn’t be

Reply:
Guess what - one plugin may stop you from entering, another from leaving - and tada, a loop